pkgs/development/tools/build-managers/apache-maven/build-package.nix

```diff
@@ -53,4 +60,8 @@ stdenv.mkDerivation (rec {
 
     runHook postBuild
   '';
-} // builtins.removeAttrs args [ "mvnFetchExtraArgs" ])
+
+  meta = args.meta or { } // {
+    platforms = args.meta.platforms or maven.meta.platforms;
```

Comment:
I am unsure if this would be correct, since `maven.meta.platforms` includes a lot more platforms than what `jre`/`jdk`/`openjdk` is stated to support as architectures.

https://github.com/NixOS/nixpkgs/blob/94afc043d55e0f170a9d8d659397db7ab152f7d4/pkgs/development/compilers/openjdk/meta.nix#L6

https://github.com/NixOS/nixpkgs/blob/30464bb5c54d8c706443f433db6777284b8a99d2/pkgs/development/tools/build-managers/apache-maven/default.nix#L47

This may be an issue of how the `maven` package itself is labelled, as I am unsure if we ship jdk packages for all of these. But this should be looked into deeper.

Reply:
This may also be a problem from `openjdk` meta attributes being too restrictive. This however I cannot confirm, as I hold no deep knowledge of why one has been chosen over the other, and primarily pointing out the differences in `meta.platforms` between two Java related derivations that should likely be equal.

Reply:
As i see you successfully built packages with maven on `aarch64-darwin`, which we wrap with `jdk` and is itself a java application. Could you see if you can run any of these built packages on `aarch64-darwin`? As it seems strange they would even be built with how `meta.platforms` is set in `openjdk`.

https://github.com/NixOS/nixpkgs/blob/30464bb5c54d8c706443f433db6777284b8a99d2/pkgs/development/tools/build-managers/apache-maven/default.nix#L30-L31

Reply:
I believe the list of Linux platforms is arbitrary, as it was last modified to support `powerpc64le-linux` in a one-line PR...

Reply:
I was also thinking it might be CI load? I thought nix would just outright refuse to build if your platform wasn't inside of `meta.platforms`... If that is not so, I am certainly learning new things. :sweat_smile:

Reply:
> Could you see if you can run any of these built packages on `aarch64-darwin`? As it seems strange they would even be built with how `meta.platforms` is set in `openjdk`.

Yes. That list actually only applies to the Linux derivations, while the darwin ones reside in `pkgs/development/compilers/openjdk/darwin`.

Reply:
> I thought nix would just outright refuse to build if your platform wasn't inside of `meta.platforms`... If that is not so, I am certainly learning new things. 😅

The package can still be built if `meta.platforms` is not defined, but Hydra will only build it on `x86_64-linux`.

Reply:
But then, any tool like `maven` being set to `lib.platforms.unix` will still build `jdk` for all `.unix` platforms as part of its build step...

I am not really sure what is going on there. Seeing as @edwtjo is the only maintainer listed for `openjdk` now, and [they have not commited to openjdk inside nixpkgs since 2016](https://github.com/NixOS/nixpkgs/commits/8c588942d0017a86a4b162c9681d2194248b322b/pkgs/development/compilers/openjdk?author=edwtjo), there might not be a good straight answer to find without some digging.

But if you (@edwtjo) know anything, please do feel free to share wisdom.

Reply:
> Yes. That list actually only applies to the Linux derivations, while the darwin ones reside in `pkgs/development/compilers/openjdk/darwin`.

Taking a peek at the toplevel, it does do this switching, so you are entirely correct.

https://github.com/NixOS/nixpkgs/blob/e6027c265a563009d536a4e36228eee179f27505/pkgs/top-level/java-packages.nix#L54-L59

Which would explain why only a certain set of supported platforms shows up when you evaluate it. But this seems flawed to base the default on as well, as it is not clear that a package you package will also be automatically supporting Darwin systems as well.

This should likely be moved to a separate issue, as it is certainly diverging from `buildMavenPackage` relevancy.